source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRRenderer.mm

```diff
@@ -226,4 +291,66 @@ + (UIView *)render:(UIView<ACRIContentHoldingView> *)view
 
     return view;
 }
+
++ (UIView *)renderInGridOrFlow:(UIView<ACRIContentHoldingView> *)view
+                      rootView:(ACRView *)rootView
+                        inputs:(NSMutableArray *)inputs
+                 withCardElems:(std::vector<std::shared_ptr<BaseCardElement>> const &)elems
+                 andHostConfig:(ACOHostConfig *)config
+{
+    ACRRegistration *reg = [ACRRegistration getInstance];
+    ACOBaseCardElement *acoElem = [[ACOBaseCardElement alloc] init];
+    ACOFeatureRegistration *featureReg = [ACOFeatureRegistration getInstance];
+
+    // Get responsive layout's host width
+    HostWidthConfig hostWidthConfig = [config getHostConfig]->getHostWidth();
+    HostWidth hostWidth = convertHostCardContainerToHostWidth([reg getHostCardContainer], hostWidthConfig);
+
+    UIView *renderedView = nil;
+
+    for (const auto &elem : elems)
+    {
+        ACRBaseCardElementRenderer *renderer =
+        [reg getRenderer:[NSNumber numberWithInt:(int)elem->GetElementType()]];
+
+        if (renderer == nil)
+        {
+            NSLog(@"Unsupported card element type:%d\n", (int)elem->GetElementType());
+            continue;
+        }
+
+        [acoElem setElem:elem];
+
+        @try
+        {
+            if ([acoElem meetsRequirements:featureReg] == NO)
+            {
+                @throw [ACOFallbackException fallbackException];
+            }
+
+            if (elem->MeetsTargetWidthRequirement(hostWidth) == false)
+            {
+                continue;
+            }
+
+            if (!elem->GetIsVisible())
+            {
+                continue;
+            }
```

Comment:
These can be one combined into one statement.

Reply:
Kept this separate intentionally for better readability of code flow.  Old flow has different handling for visibility updates, this gives 1:1 match with code branching in old vs new render flow. 
Will you prefer it to be combined?